backend/user_manager.js

```diff
@@ -64,14 +64,35 @@ module.exports = function (app, config, callback) {
 
     app.post('/user/add', checkLogin, checkAdmin, function (req, res) {
         const email = req.body.email;
-        const pass = md5(req.body.password);
-        mysqlUtils.createUser(email, pass, config.mysqlConnPool, function (err, isSuccessful) {
-            if (!err && isSuccessful) {
-                res.redirect('/dashboard');
-            } else {
-                res.render('login.pug', { message: 'Unable to Create User' });
-            }
-        });
+        const username = req.body.username.trim();
+        const unHashedPass = req.body.password;
+        const pass = md5(unHashedPass);
+
+        if (username.length > 30) {
+            return res.status(406).send('Username length exceeds 30 characters.');
+        } else if (username.length == 0) {
```

Comment:
Can we create a function for validation as we have made `validateEmail` used below?

Reply:
done in the latest commit.